includes/Admin/Settings_Page.php

```diff
@@ -0,0 +1,270 @@
+
+/**
+ * Manages the admin settings page for AI experiments.
+ *
+ * @since 0.1.0
+ */
+class Settings_Page {
+
+	/**
+	 * The experiment registry instance.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var \WordPress\AI\Experiment_Registry
+	 */
+	private Experiment_Registry $registry;
+
+	/**
+	 * The option name for the global experiments toggle.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var string
+	 */
+	private const GLOBAL_OPTION = 'ai_experiments_enabled';
+
+	/**
+	 * The settings page slug.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @var string
+	 */
+	private const PAGE_SLUG = 'ai-experiments';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param \WordPress\AI\Experiment_Registry $registry The experiment registry.
+	 */
+	public function __construct( Experiment_Registry $registry ) {
+		$this->registry = $registry;
+	}
+
+	/**
+	 * Initializes the settings page hooks.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return void
+	 */
+	public function init(): void {
+		add_action( 'admin_menu', array( $this, 'register_menu' ) );
+		add_action( 'admin_init', array( $this, 'register_settings' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_styles' ) );
+	}
+
+	/**
+	 * Registers the admin menu item.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return void
+	 */
+	public function register_menu(): void {
+		add_options_page(
+			__( 'AI Experiments', 'ai' ),
+			__( 'AI Experiments', 'ai' ),
+			'manage_options',
+			self::PAGE_SLUG,
+			array( $this, 'render_page' )
+		);
+	}
+
+	/**
+	 * Registers all settings for experiments.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @return void
+	 */
+	public function register_settings(): void {
+		// Register the global toggle.
+		register_setting(
+			'ai_experiments',
+			self::GLOBAL_OPTION,
+			array(
+				'type'              => 'boolean',
+				'default'           => false,
+				'sanitize_callback' => 'rest_sanitize_boolean',
+			)
+		);
+
+		// Register settings for each experiment.
+		foreach ( $this->registry->get_all_experiments() as $experiment ) {
+			$experiment_id     = $experiment->get_id();
+			$experiment_option = "ai_experiment_{$experiment_id}_enabled";
+
+			register_setting(
+				'ai_experiments',
+				$experiment_option,
+				array(
+					'type'              => 'boolean',
+					'default'           => false,
+					'sanitize_callback' => 'rest_sanitize_boolean',
+				)
+			);
+
+			// Allow experiments to register their own custom settings.
+			if ( ! method_exists( $experiment, 'register_settings' ) ) {
+				continue;
+			}
+
+			$experiment->register_settings();
+		}
+	}
+
+	/**
+	 * Enqueues styles for the settings page.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param string $hook_suffix The current admin page hook suffix.
+	 * @return void
+	 */
+	public function enqueue_styles( string $hook_suffix ): void {
+		if ( 'settings_page_' . self::PAGE_SLUG !== $hook_suffix ) {
+			return;
+		}
+
+		// Enqueue WordPress components styles for block editor UI.
+		wp_enqueue_style( 'wp-components' );
```

Comment:
No need to enqueue this, since it's already a dependency of the style below. It'll be automatically enqueued.

Reply:
Resolved in 1aba23d